### PR TITLE
Hiding Entity class away, in favor of a public IEntity interface

### PR DESCRIPTION
**Pull request by GenerelSchwerz**

### CODE REVIEW

**Comment by u9g** [comment]
can you name the pr something meaningful

**Comment by GenerelSchwerz** [comment]
done

**Review by u9g — APPROVED** [review]

**Comment by u9g** [comment]
U9G — 03/01/2022: 
> why did you call it IEntity? its literally an entity class

Generel — 03/01/2022: 
> When its exposed like that, it can't be used as a class since it isn't defined until you return a value from the loader function
```js
import loader, {Entity} from "prismarine-entity";
```
> so getting it like this returns undefined
not a static reference of a class

File: types/entity.d.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/// <reference types="vec3" />
+/// <reference types="prismarine-item" />
+
+import {EventEmitter} from 'events';
+import { Vec3 } from 'vec3';
+import { Item } from 'prismarine-item';
+import { ChatMessage } from 'prismarine-chat';
+
+
+export class Entity extends EventEmitter {
```
Comment (u9g) [inline types/entity.d.ts:11]: is there any reason you brought this to it's own file rather than just leaving it in the index.d.ts?

**Comment by rom1504** [comment]
old

**Pull request closed without merging**